contracts/__test__/contracts.test.ts

```diff
@@ -3253,6 +3254,266 @@ describe('reti', () => {
         })
     })
 
+    describe('APRTest', () => {
+        let validatorId: number
+        let validatorOwnerAccount: Account
+        let poolAppId: bigint
+        let firstPoolKey: ValidatorPoolKey
+        let ourPoolClient: StakingPoolClient
+        const epochRoundLength = 1
+
+        const stakeAmount = 10000n.algo()
+
+        function wait(ms: number): Promise<void> {
+            return new Promise((resolve) => setTimeout(resolve, ms))
+        }
+
+        // add validator and 1 pool for subsequent stake tests
+        beforeAll(async () => {
+            // let's make sure 'rounds per day' is always a small number by having a LOT of time between blocks
+            await fixture.context.algod.setBlockOffsetTimestamp((60 * 60 * 24) / 10).do()
+            // have at least 12 dummy transactions so contract will set roundsPerDay from chain
+            await incrementRoundNumberBy(fixture.context, 12)
+            // wait 3 seconds to break param cache (disabling seems to break other things)
+            await wait(3000)
+
+            // Fund a 'validator account' that will be the validator owner.
+            validatorOwnerAccount = await fixture.context.generateAccount({
+                initialFunds: AlgoAmount.Algos(500),
+                suppressLog: true,
+            })
+            consoleLogger.info(`validator account ${validatorOwnerAccount.addr}`)
+
+            const config = createValidatorConfig({
+                owner: validatorOwnerAccount.addr.toString(),
+                manager: validatorOwnerAccount.addr.toString(),
+                validatorCommissionAddress: validatorOwnerAccount.addr.toString(),
+                minEntryStake: AlgoAmount.Algos(1000).microAlgos,
```

Comment:
Does the contact have a lower limit for `minEntryStake`? I'm missing a lot of detailed context here but since this bug was caused by low stake maybe we want to test with 1? Probably will never happen, but if it's possible might feel better testing the "worst case"

Reply:
keep in mind a normal pool should never have <30K otherwise it's earning nothing..
in my synthetic test, I put in 10K algo as the stake and then send payments directly to the reti pool so that its treated as part. awards - sending out 6% of 10K / 365 or 1.643835 algo per virtual day.
I have to game the timestamps and seed a bunch of rounds prior to the pool being first added so that a 'day' is just 10 rounds, then I send the 1.64xx algo, call epoch update, and do stub txns to 'end the day'.  Repeat a couple times.
APR should be approximately 6% and it ends up being 5.96 or so (partially because the stake is the 'average' stake over a day)